### PR TITLE
add documentation for testing API

### PR DESCRIPTION
**Pull request by kitsonk**

### CODE REVIEW

File: configuration.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Configuration reference
+
+Deno is a flexible development tool and runtime. Because of that there are
+numerous configuration options which are supported. This page is a reference for
+this options.
```
Comment (dsherret) [inline configuration.md:5]:
```suggestion
these options.
```

File: configuration.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+The following top level keys are supported:
+
+- `"compilerOptions"` - Options that are used when transpiling or type checking
+  TypeScript and JavaScript. This is a strict subset of the
+  [TSconfig `"compilerOptions"` options](https://www.typescriptlang.org/tsconfig).
+- `"importMap"` - Identifies an import map to be applied when resolving imports.
+- `"lint"` -
+- `"fmt"` -
+- `"tasks"` -
```
Comment (dsherret) [inline configuration.md:43]: Missing here and below.

File: configuration.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+There are some options that only impact how `vscode_deno` integrates with the
+Deno language server. Other clients may or may not use these.
+
+## Pragmas
+
+Deno supports various pragmas within code which can have various effects on how
+the code is handled.
+
+### `@deno-types`
```
Comment (dsherret) [inline configuration.md:66]: All these.

File: language_server/testing_api.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
+## Testing API
+
+The Deno language server supports a custom set of APIs to enable testing. These
+are built around providing information to enabled the
```
Comment (dsherret) [inline language_server/testing_api.md:4]:
```suggestion
are built around providing information to enable the
```

File: language_server/testing_api.md [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+When a version of Deno that supports the testing API encounters a client which
+supports the capability, it will initialize the code which handles the test
+detection and will start providing the notifications which enable it.
+
+It should also be noted that when the testing API capabilities are enabled, the
+testing code lenses will no longer be sent to the client.
+
+### Settings
+
+There are specific settings which change the behavior language server:
```
Comment (dsherret) [inline language_server/testing_api.md:38]:
```suggestion
There are specific settings which change the behavior of the language server:
```

File: language_server/testing_api.md [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+
+- `deno.testing.args` - An array of strings which will be provided as arguments
+  when executing tests. Working in the same fashion as the `deno test`
```
Comment (dsherret) [inline language_server/testing_api.md:41]:
```suggestion
  when executing tests. This works in the same fashion as the `deno test`
```

File: language_server/testing_api.md [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+  subcommand.
+- `deno.testing.enable` - a binary flag that enables or disables the testing
```
Comment (dsherret) [inline language_server/testing_api.md:43]:
```suggestion
- `deno.testing.enable` - A binary flag that enables or disables the testing
```

Or other one above lower case.

File: language_server/testing_api.md [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+```
+
+### Requests
+
+The server handles two different requests:
+
+#### `deno/testRun`
+
+To request the language server to perform a set of tests, the client sends a
+`deno/testRun` request, which includes that ID of the test run to be used in
```
Comment (dsherret) [inline language_server/testing_api.md:268]:
```suggestion
`deno/testRun` request, which includes the ID of the test run to be used in
```

File: vscode_deno/testing_api.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+Based on other settings that you have, those options will be automatically
+merged into the "command line" used when running tests unless explicitly
+provided in the _Deno > Testing: Args_ setting. For example if you have a _Deno:
+Import Map_ (`deno.importMap`) set, the value of that will be used unless you
+have provided an explicit `--import-map` value in the testing args setting.
+
+### Known limitations and caveats
+
+Because of the way the Deno test runner runs, it is not possible to exclude (or
+explicitly include) as test step. While the vscode UI will allow you to do this,
```
Comment (dsherret) [inline vscode_deno/testing_api.md:81]:
```suggestion
explicitly include) a test step. While the vscode UI will allow you to do this,
```

**Review by dsherret — COMMENTED** [review]
Almost there.

Was the configuration.md file meant to be added? Seems like it's missing some parts.

**Review by UltiRequiem — APPROVED** [review]

**Comment by kitsonk** [comment]
Yes, configuration.md was not intended to be included.

I have addressed the rest of the feedback.

**Review by dsherret — APPROVED** [review]
LGTM. Sorry for my delay.

**Pull request merged**